src/Sparrow/Json/BlittableWriter.cs

```diff
@@ -311,9 +313,13 @@ public unsafe int WriteVariableSizeLong(long value)
             // https://developers.google.com/protocol-buffers/docs/encoding?csw=1#types
             // for negative values
 
-            var buffer = stackalloc byte[10];
-            var count = 0;
-            var v = (ulong)((value << 1) ^ (value >> 63));
+            if (_allocated == null)
+                _allocated = _context.GetMemory(10);
+
```

Comment:
What is the error? and are you sure you aren't leaking this / reusing?
Also, why not do the stack alloc here, should be faster, and it is fixed & very small